auth0/v3/rest.py

```diff
@@ -105,6 +111,12 @@ def __init__(self, jwt, telemetry=True, timeout=5.0, options=None):
         # Cap the maximum number of retries to 10 or fewer. Floor the retries at 0.
         self._retries = min(self.MAX_REQUEST_RETRIES(), max(0, options.retries))
 
+        # Modify default 1s maximum delay
+        self._delay = max(
```

Comment:
We don't need a hard max, happy to just have this parameterised and default to 1000

Can you remove `MAX_REQUEST_RETRY_DELAY`, change the new `RestClientOptions` keyword arg to `max_delay` then just change this to:

```py
self._delay = options.max_delay
```